File: codigo/backend/go-backend/models/produto.go
package models

import "fmt"


type Produto struct {
    ID           uint    `gorm:"primaryKey"`
    URL          string  `gorm:"type:varchar(255)"`
    Descricao    string  `gorm:"type:text"`
    Especificacao string `gorm:"type:text"`
    Preco        float64 `gorm:"type:numeric(10,2)"`
}


func (Produto) TableName() string {
    return "produtos" // Nome da tabela no banco de dados
}

func (p *Produto) ToMap() map[string]interface{} {
    return map[string]interface{}{
        "id":           p.ID,
        "url":          p.URL,
        "descricao":    p.Descricao,
        "especificacao": p.Especificacao,
        "preco":        fmt.Sprintf("%.2f", p.Preco),  // Convert to string for JSON serialization
    }
}

func (p *Produto) String() string {
    return fmt.Sprintf("<Produto %d>", p.ID)
}

File: codigo/backend/sap-ecc-emulator/src/models/produto.py
from db import db

class Produto(db.Model):
    __tablename__ = 'produtos'

    id = db.Column(db.Integer, primary_key=True)
    url = db.Column(db.String(255))
    descricao = db.Column(db.Text)
    especificacao = db.Column(db.Text)
    preco = db.Column(db.Numeric(10, 2))

    def __repr__(self):
        return f'<Produto {self.id}>'

    def to_dict(self):
        return {
            'id': self.id,
            'url': self.url,
            'descricao': self.descricao,
            'especificacao': self.especificacao,
            'preco': str(self.preco)  # Convert to string for JSON serialization
        }
